table/arrow_utils.go

```diff
@@ -1250,5 +1566,56 @@ func dataFileStatsFromParquetMetadata(pqmeta *metadata.FileMetaData, statsCols m
 		nullValueCounts: nullValueCounts,
 		nanValueCounts:  nanValueCounts,
 		splitOffsets:    splitOffsets,
+		colAggs:         colAggs,
+	}
+}
+
+func parquetFilesToDataFiles(fileIO iceio.IO, meta *MetadataBuilder, paths iter.Seq[string]) iter.Seq2[iceberg.DataFile, error] {
```

Comment:
I'm reluctant to add this with this PR. Do we handle the name-mapping properly?

Reply:
yup, name mapping is handled properly (as far as I can tell via testing) though over the course of writing tests for the subsequent changes, I did find a couple edge cases which I fixed in those commits. They'll be included in the next PR after this one.